frb_codegen/src/config.rs

```diff
@@ -242,6 +240,72 @@ pub fn parse(raw: RawOpts) -> Vec<Opts> {
         .collect()
 }
 
+fn get_refined_c_output(
```

Comment:
last uber-nit: would be great to add unit tests for this function, since it is so isolated and has a bit of complexity.

for example:

1. single api file, no extra
2. single api file, has extra
3. multi api files, no extra
4. multi api files, has extra

Reply:
Tests for 1,3 and 4 have already existed, so I only add tests for 2 in 47bbcec14a175d1490d0f0fb7d1fc8195f3c8f8c.
And for all examples, I didn't add case with flag `extra-c-output-path` but without the flag `c-output`, in that:
1. this case is quite rare in use;
2. with this case added, there would be new files produced with arbitrary names after running the test both locally by `just precommit` or online by CI test. It is difficult to check correctness.

Reply:
Well I mean `unit test` - https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html. The unit tests are a good way to ensure this function never breaks, and also documents the function (by providing examples)

The 47bbcec end-to-end test is not required (but if you like surely you can keep it)

Reply:
> Well I mean `unit test` - https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html. The unit tests are a good way to ensure this function never breaks, and also documents the function (by providing examples)
> 
> The [47bbcec](https://github.com/fzyzcjy/flutter_rust_bridge/commit/47bbcec14a175d1490d0f0fb7d1fc8195f3c8f8c) end-to-end test is not required, but if you like surely you can keep it, as long as it does not cause problems

Actually, before adding test for CI and justfile, I've come up the unit test idea for `pure_dart` and `pure_dart_multi`.  But question is I didn't find a way to test following the procedure in `build.rs` with no syntax error, say I create path `tests\test.rs` at the same level of `src`/`build.rs` in project `pure_dart`, the content like this:
```rust
#[path = "../build.rs"]
// mod module_name;
use lib_flutter_rust_bridge_codegen::{
    config_parse, frb_codegen, get_symbols_if_no_duplicates, RawOpts,
};

#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn test_1() {
        println!("cargo:rerun-if-changed={}", RUST_INPUT);
        // Options for frb_codegen
        let mut raw_opts = RawOpts {
            // Path of input Rust code
            rust_input: vec![RUST_INPUT.to_string()],
            // Path of output generated Dart code
            dart_output: vec![DART_OUTPUT.to_string()],
            wasm: true,
            dart_decl_output: Some("../dart/lib/bridge_definitions.dart".into()),
            dart_format_line_length: 120,
            // for other options use defaults
            ..Default::default()
        };

        // ...

        // generation of rust api for ffi
        let all_symbols = get_symbols_if_no_duplicates(&configs).unwrap();
        for config in configs.iter() {
            frb_codegen(config, &all_symbols).unwrap();
        }
    }

    #[test]
    fn test_2() {}
}
```
I don't know how to make it compliable.

Reply:
Well for unit test I mean unit test the single function - `get_refined_c_output`. Just test this one function, and pretend the rest of all code *does not exist at all*. (That's why it is called "unit" testing - only test the unit, i.e. the function)

Reply:
Ok, I found the way with no syntax error, just change the head into this:
```rust
#[path = "../build.rs"]
mod lib_flutter_rust_bridge_codegen;
use lib_flutter_rust_bridge_codegen::{
    config_parse, frb_codegen, get_symbols_if_no_duplicates, RawOpts,
};
```
Then it is compilable. 
In addition, if this is not the unit test you want, please tell me.

Reply:
> Well for unit test I mean unit test the single function - `get_refined_c_output`. Just test this one function, and pretend the rest of all code _does not exist at all_. (That's why it is called "unit" testing - only test the unit, i.e. the function)

I see. But I wonder how to catch panic for:
```rust
fn bail(err: clap::ErrorKind, message: Cow<str>) {
    RawOpts::command().error(err, message).exit()
}
```
The `bail` would be called in a `should_panic` test.

Reply:
LGTM, if you want to test it just catch the panic

Reply:
I don't know how, since

> LGTM, if you want to test it just catch the panic

I mean I don't know how to catch the panic from ` RawOpts::command().error(err, message).exit()`. From commit aa11a608bf2f0b801b2d306ec6a636202f9fef66, if uncomment test case `test_coutput_with_inconsistent_number_of_input_block_api`, then the test routine can't proceed.